src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -352,9 +399,12 @@ public int compare(WordHolder h1, WordHolder h2)
 
           i.advanceTo(itr.wordsWalked);
           if (i.hasNext()) {
-            wordsToAdd.add(new WordHolder(i.next(), i));
+            i.word = i.next();
+            qIndex++;
+          } else {
+            removeElement(theQ, qIndex, qSize);
```

Comment:
@egrs perf diff should be negligible because the first element of the queue is usually removed just a few times during the whole intersection or union operation, while sort is performed thousands of times.